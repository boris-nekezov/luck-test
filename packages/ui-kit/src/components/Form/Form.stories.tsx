import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Form } from './Form';

import { Primary } from '../Input/Input.stories';

export default {
  title: 'Form',
  component: Form,
} as ComponentMeta<typeof Form>;

const Template: ComponentStory<typeof Form> = (args) => (
  <Form {...args}>
    <Primary {...Primary.args} />
    <Primary {...Primary.args} />
    {/*// todo replace it with button component */}
    <button>Submit</button>
  </Form>
);

export const FormBasic = Template.bind({});