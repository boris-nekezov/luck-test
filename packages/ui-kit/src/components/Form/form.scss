@import '../../styles/app.scss';

.form {
	@include theme-switcher() {
		background: $form-background--light;
	}

	padding: $space-md;
	margin: $space-md;
	border-radius: $space-sm;
	background: $form-background--dark;

	.input {
		margin-bottom: $space-md;
	} 

	fieldset {
		border: 0;
	}
}
