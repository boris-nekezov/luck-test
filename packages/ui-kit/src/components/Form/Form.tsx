import './form.scss';

export interface FormProps {
  handleSubmit: (e: any) => void;
  children: React.ReactNode,
}

export const Form = (props: FormProps) => {
  const {
    handleSubmit,
    children,
  } = props;

  return (
    <div className='form'>
      <form
        onSubmit={handleSubmit}
        method="post"
      >
        <fieldset>
          <legend>
            {children}
          </legend>
        </fieldset>
      </form>
    </div>
  )
}
