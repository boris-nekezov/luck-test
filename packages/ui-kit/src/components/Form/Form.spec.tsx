import React from 'react';
import { render } from '@testing-library/react';
import { Form } from './Form';

describe('Given `Form` component', () => {
  const mockFn = jest.fn();
  let wrapper: ReturnType<typeof render>;


  describe('when it is rendered', () => {
    beforeEach(() => {
      wrapper = render(<Form
        handleSubmit={mockFn}>
        <h2>Some form</h2>
      </Form>);
    });

    it('should match snapshot', () => {
      expect(wrapper.baseElement).toMatchSnapshot();
    })
  })
});
