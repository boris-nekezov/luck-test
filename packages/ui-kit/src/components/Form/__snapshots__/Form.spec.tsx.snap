// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given \`Form\` component when it is rendered should match snapshot 1`] = `
<body>
  <div>
    <div
      class="form"
    >
      <form
        method="post"
      >
        <fieldset>
          <legend>
            <h2>
              Some form
            </h2>
          </legend>
        </fieldset>
      </form>
    </div>
  </div>
</body>
`;
